### PR TITLE
Drivers: Pinmux: Add CAN pins for STM32 F4 MCUs

### PR DESCRIPTION
**Pull request by Benichou34**
Add CAN pins for STM32 F4 MCUs, in pinmux_stm32f4.h

### CODE REVIEW

**Review by erwango — COMMENTED** [review]
@Benichou34, can you provide the reference of the F4 you're working with?
I've had a look to some ref (f429) and this does not cover all the entries you've put there.

Besides, for now, in zephyr tree, there is no CAN enabled (either in dts, fixup, board) on STM32F4.
Have you been able to test it successfully on the reference you're using?
If yes, it would make sense to add these along with the pin entries, for consistency.
Could you consider to do the upstream work for enabling CAN on  family, as part of a larger commit?
Would be great.

File: drivers/pinmux/stm32/pinmux_stm32f4.h [hunk 1]
```diff
@@ -83,6 +83,8 @@
 	(STM32_PINMUX_ALT_FUNC_4 | STM32_OPENDRAIN_PULLUP)
 #define STM32F4_PINMUX_FUNC_PA8_UART7_RX    \
 	(STM32_PINMUX_ALT_FUNC_8 | STM32_PUSHPULL_NOPULL)
+#define STM32F4_PINMUX_FUNC_PA8_CAN3_RX \
```
[the remark this answers is not in the document]
Comment (Benichou34) [reply drivers/pinmux/stm32/pinmux_stm32f4.h]: We have our own board with a STM32F413 MCU. We also use some Nucleo-F413ZH dev board.
We made tested it successfully with this Nucleo board and two brand of CAN transceivier (TJA1050 and MCP2551).

We modified Nucleo-F413 dts/fixup/pinmux for add CAN support, and we plan to add it in a future pull request.

[the remark this answers is not in the document]
Comment (erwango) [reply drivers/pinmux/stm32/pinmux_stm32f4.h]: Ok, so let's wait next pull request enabling CAN on at least one F4 board.

**Review by erwango — CHANGES_REQUESTED** [review]
For coherency, we'll wait for CAN soc support first.

**Pull request closed without merging**